lib/project.ts

```diff
@@ -358,23 +359,77 @@ export class Project implements Project.IProject {
 
 	public printProjectProperty(property: string): IFuture<void> {
 		return (() => {
-			if(!property) {
-				var names = _.keys(this.projectData);
-				var sortedProperties = _.sortBy(names, (propertyName: string) => {
-					return propertyName.toLowerCase();
-				});
-				_.each(sortedProperties, (propertyName: string) => this.$logger.out(propertyName + ": " + this.projectData[propertyName]));
-				return;
+			if(this.projectData) {
+				var schema: any = this.getProjectSchema().wait();
+
+				if(property) {
+					var normalizedPropertyName = this.$projectPropertiesService.normalizePropertyName(property, this.projectData);
+
+					if(options.validValue) {
+						// '$ appbuilder prop print <PropName> --validValue' called inside project dir
+						var prop: any = schema[normalizedPropertyName];
+						this.printValidValuesOfProperty(prop).wait();
+					} else {
+						// '$ appbuilder prop print <PropName>' called inside project dir
+						if(_.has(this.projectData, normalizedPropertyName)) {
+							this.$logger.out(this.projectData[normalizedPropertyName]);
+						} else {
+							this.$errors.fail("Unrecognized project property '%s'", property);
+						}
+					}
+				} else {
+					if(options.validValue) {
+						// 'appbuilder prop print --validValue' called inside project dir
+						_.each(_.keys(schema), propKey => {
+							var prop = schema[propKey];
+							this.$logger.info("  " + propKey);
+							this.printValidValuesOfProperty(prop).wait();
+						});
+					} else {
+						// 'appbuilder prop print' called inside project dir
+						var propKeys: any = _.keys(this.projectData);
+						var sortedProperties = _.sortBy(propKeys, (propertyName: string) => {
+							return propertyName.toLowerCase();
```

Comment:
Well, the idea is to persist the prioperties as they are, but to sort them based on their toLowerCase values. That's why toLowerCase is inside the sortBy